modules/geochemistry/src/utils/GeochemicalDatabaseReader.C

```diff
@@ -27,6 +28,11 @@ GeochemicalDatabaseReader::read(FileName filename)
   // Read the JSON database
   std::ifstream jsondata(filename);
   jsondata >> _root;
+
+  // Validate the JSON database so that we don't have to check array sizes,
+  // check for conversion issues, etc.
+  GeochemicalDatabaseValidator dbv(filename, _root);
+  dbv.validate();
```

Comment:
Yea, i like validating when reading.  Better than a standalone validator, which is what i thought you were going to do

Reply:
It seemed more useful to do it when running so we can give a useful error.